packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -774,21 +760,52 @@ export class ContainerRuntime
 
 		const mc = loggerToMonitoringContext(logger);
 
+		// Some options require a minimum version of the FF runtime to operate, so the default configs will be generated
+		// based on the compatibility mode.
+		// For example, if compatibility mode is set to "1", the default configs will ensure compatibility with FF runtime
+		// 1.x  or later. If the compatibility mode is set to "2", the default values will be generated to ensure compatibility
+		// with FF runtime 2.x or later.
+		// Our policy is to support N/N-1 compatibility by default, where N is the most recent public major release of the runtime.
+		// Therefore, if the customer does not provide a compatibility mode, we will default to use N-1.
+		const defaultCompatibilityMode = "1"; // TODO: Automatically set to be N-1.
+		const compatibilityMode = runtimeOptions.compatibilityMode ?? defaultCompatibilityMode;
+		const defaultConfigs = getConfigsForCompatMode(compatibilityMode);
+
+		// Here we assign each container runtime option to the value provided in `runtimeOptions`.
```

Comment:
So do we allow someone to explicitly set a value that's "not allowed" per the compat mode stuff?  I'm thinking it would be better to fail fast if that happens.  But it may be difficult to specify in code how to enforce that.

Reply:
I think we have a few options here:
1. We can throw an error and fail fast.
1. We can "overrule" the configuration with what's "correct". There is some precedent for this in today's code. We currently we will automatically disable compression (without an error or warning) if grouped batching is disabled, regardless of what the user did.
1. Keep the status quo and allow customers to set their own options. `DocumentSchema` should prevent data corruption scenarios but could result in confusion from the customer's perspective if clients are disallowed from joining.

Reply:
Yeah I've been thinking about this reading through this a second time.  I think it's really hard to get right, it might even require a custom function to be added to the `runtimeOptionConfigs` entry for each config setting to validate that a given value properly aligns with the "disabled"/"legacy" case.  Probably not worth it, as long as we do have confidence that the runtime will fail fast in all cases.  That's what I'm not sure about.